demo/lightning/text-based/distributed/model.py

```
@@ -0,0 +1,309 @@
+                    bucket_name=self.bucket_name,
+                    objects=self.objects[start:end],
+                    storage_client=self.storage_client,
+                    dataflux_download_optimization_params=self.
+                    dataflux_download_optimization_params,
+                    retry_config=self.config.download_retry_config,
+            ):
+                table = self.data_format_fn(bytes_content)
+                for batch in table.iter_batches(batch_size=self.batch_size,
+                                                columns=self.columns):
+                    yield from batch.to_pylist()
+
+
+def main():
+    args = parse_args()
+    logging.basicConfig(level=args.log_level)
+    # If running locally, default to localhost port 1234, and use manually specified rank.
+    if args.local:
+        logging.info(
+            "Executing local configuration. Setting MASTER_ADDR/PORT to localhost:1234"
+        )
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["WORLD_SIZE"] = str(args.num_nodes)
+        os.environ["MASTER_PORT"] = "1234"
+        os.environ["NODE_RANK"] = str(args.rank)
+    else:
+        # Set environment variables for distributed workload.
+        init_processes()
+    # The prefix passed here determines which parquet files are read from our bucket.
+    # Listing results must be sorted to guarantee each worker gets a different fileset.
+    config = df_iter.Config(prefix=args.prefix, sort_listing_results=True)
+    my_model = TextDemoModel(encoder, decoder)
+    bsize = args.batch_size
+
+    # Construct the lightning trainer and run the fit with our model and custom dataset.
+    # Note that limit_train_batches specifies how much of the dataset to train each epoch.
+    trainer = pl.Trainer(accelerator='cpu',
+                         devices=args.devices,
+                         strategy=pl.strategies.DDPStrategy(
+                             timeout=datetime.timedelta(minutes=30)),
```

Can you update this to a smaller timeout here so that it fails faster if something is wrong and doesn't appear to be stuck?